### PR TITLE
Fix Tomcat URL handling fallout

### PR DESCRIPTION
**Pull request by thomasmey**
Issue https://github.com/flyway/flyway/issues/1549
Untested proposed fix.
Patch against 4.1.2

### CODE REVIEW

File: flyway-core/src/main/java/org/flywaydb/core/internal/util/scanner/classpath/ClassPathScanner.java [hunk 1]
```diff
@@ -308,11 +308,16 @@ private ClassPathLocationScanner createLocationScanner(String protocol) {
...
         if ("jar".equals(protocol)
-                || "war".equals(protocol)
+                || "war".equals(protocol) //Tomcat
                 || "zip".equals(protocol) //WebLogic
                 || "wsjar".equals(protocol) //WebSphere
                 ) {
-            JarFileClassPathLocationScanner locationScanner = new JarFileClassPathLocationScanner();
+            JarFileClassPathLocationScanner locationScanner;
+            if ("war".equals(protocol)) {
+                locationScanner = new JarFileClassPathLocationScanner("*/");
```
Comment (axelfontaine) [inline flyway-core/src/main/java/org/flywaydb/core/internal/util/scanner/classpath/ClassPathScanner.java]: Won't this be breaking older Tomcat releases?

Comment (thomasmey) [inline flyway-core/src/main/java/org/flywaydb/core/internal/util/scanner/classpath/ClassPathScanner.java]: I don't think so, as older tomcat version will return URLs like jar:xxx!/ for war files, and new tomcats will return URLs like war:xxx*/. but somebody should test this...

Comment (thomasmey) [inline flyway-core/src/main/java/org/flywaydb/core/internal/util/scanner/classpath/ClassPathScanner.java]: Hi, I manually did test this patch against Tomcat 8.5.5 and 8.5.13 (old and new URL handling) with plain 4.1.2 and patched 4.1.2.
The patched 4.1.2 always works correctly in all four constellations.
Do you need anything else to merge this PR?

Comment (axelfontaine) [inline flyway-core/src/main/java/org/flywaydb/core/internal/util/scanner/classpath/ClassPathScanner.java]: OK, works for me. Thanks for verifying!

**Comment by axelfontaine** [comment]
@thomasmey Could you please get in touch with me by email? (axel at boxfuse.com) Thanks!

**Pull request merged**